src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/TestHooks/ValueTaskExtensions.cs

```diff
@@ -0,0 +1,241 @@
+
+using System;
+using System.Runtime.CompilerServices;
+using System.Runtime.ExceptionServices;
+using System.Threading.Tasks;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.Shared.TestHooks
+{
+    internal static class ValueTaskExtensions
+    {
+        /// <summary>
+        /// Returns an awaitable for the specified task that will never throw, even if the source task
+        /// faults or is canceled.
+        /// </summary>
+        /// <param name="task">The task whose completion should signal the completion of the returned awaitable.</param>
+        /// <param name="captureContext">if set to <c>true</c> the continuation will be scheduled on the caller's context; <c>false</c> to always execute the continuation on the threadpool.</param>
+        /// <returns>An awaitable.</returns>
+        public static NoThrowValueTaskAwaitable NoThrowAwaitableInternal(this ValueTask task, bool captureContext = true)
+        {
+            return new NoThrowValueTaskAwaitable(task, captureContext);
+        }
+
+        /// <summary>
+        /// Returns an awaitable for the specified task that will never throw, even if the source task
+        /// faults or is canceled.
+        /// </summary>
+        /// <param name="task">The task whose completion should signal the completion of the returned awaitable.</param>
+        /// <param name="captureContext">if set to <c>true</c> the continuation will be scheduled on the caller's context; <c>false</c> to always execute the continuation on the threadpool.</param>
+        /// <returns>An awaitable.</returns>
+        public static NoThrowValueTaskAwaitable<TResult> NoThrowAwaitableInternal<TResult>(this ValueTask<TResult> task, bool captureContext = true)
+        {
+            return new NoThrowValueTaskAwaitable<TResult>(task, captureContext);
+        }
+
+        /// <summary>
+        /// An awaitable that wraps a task and never throws an exception when waited on.
+        /// </summary>
+        public readonly struct NoThrowValueTaskAwaitable
+        {
+            /// <summary>
+            /// The task.
+            /// </summary>
+            private readonly ValueTask _task;
+
+            /// <summary>
+            /// A value indicating whether the continuation should be scheduled on the current sync context.
+            /// </summary>
+            private readonly bool _captureContext;
+
+            /// <summary>
+            /// Initializes a new instance of the <see cref="NoThrowValueTaskAwaitable" /> struct.
+            /// </summary>
+            /// <param name="task">The task.</param>
+            /// <param name="captureContext">Whether the continuation should be scheduled on the current sync context.</param>
+            public NoThrowValueTaskAwaitable(ValueTask task, bool captureContext)
+            {
+                Contract.ThrowIfNull(task, nameof(task));
+                _task = task.Preserve();
```

Comment:
doc

Reply:
➡️ Rewrote this to use a direct copy of the code from vs-threading (only changes were renames to avoid code style warnings):
https://github.com/microsoft/vs-threading/blob/main/src/Microsoft.VisualStudio.Threading/TplExtensions.cs